app/services/authentication/authenticator.rb

```diff
@@ -26,24 +26,30 @@ def self.call(*args)
 
     def call
       identity = Identity.build_from_omniauth(provider)
-
       return current_user if current_user_identity_exists?
 
-      user = proper_user(identity)
-      user = if user.nil?
-               build_user
-             else
-               update_user(user)
-             end
+      ActiveRecord::Base.transaction do
+        user = proper_user(identity)
+        user = if user.nil?
+                 build_user!
+               else
+                 update_user(user)
+               end
 
-      save_identity(identity, user)
+        identity.user = user if identity.user_id.blank?
+        new_record = identity.new_record?
+        identity.save!
+        record_identity_creation(identity) if new_record
 
-      user.skip_confirmation!
+        user.skip_confirmation!
 
-      flag_spam_user(user) if account_less_than_a_week_old?(user, identity)
+        flag_spam_user(user) if account_less_than_a_week_old?(user, identity)
 
-      user.save!
-      user
+        user.save!
+        user
+      end
+    rescue StandardError => e
+      Rails.logger.error(e)
```

Comment:
I think in order to maintain parity with what this service was doing before, I have to return `user` here from direction in the transaction...but I wonder if we actually really _need_ to return it here?

Also, I just logged any errors that pop up here (this would catch any kind of validation errors from when we `save!` either the `identity` or the `user`.

Reply:
Why don't we want these errors bubbling up? Seems like the only reason to use `save!` is so we raise an error

Reply:
I guess that's a fair point, I just wasn't sure if we wanted to let it explode in that way or know about it via logging. But I think I'll just remove the logger.